app/public/index.html

```diff
@@ -1,14 +1,107 @@
+      crossorigin="anonymous" />
+  </head>
+  <body>
     <!-- Richard: This will send a message to the websocket server -->
-    <button id="send">Send message to client</button>
-</body>
-<script src="./constants.js"></script>
-<script src="./script.js"></script>
+    <div class="login-container">
+      <h2>Login</h2>
+      <form id="loginForm">
+        <div class="input-group">
+          <label for="username">Username</label>
+          <input
+            type="text"
+            id="username"
+            name="username"
+            placeholder="Username"
+            required />
+        </div>
+        <div class="input-group">
+          <label for="password">Password</label>
+          <input
+            type="password"
+            id="password"
+            name="password"
+            placeholder="Password"
+            required />
+        </div>
+        <button type="submit" id="submit">Login</button>
+        <div id="error-message" style="color: red"></div>
+      </form>
+    </div>
+
+    <!-- <button id="send">Send message to client</button> -->
+    <div class="create-container">
+      <h2>Create Account</h2>
+      <form id="createAccountForm">
+        <div class="input-group">
+          <label for="newUsername">Username</label>
+          <input
+            type="text"
+            id="newUsername"
+            name="newUsername"
+            placeholder="Username"
+            required />
+        </div>
+        <div class="input-group">
+          <label for="newPassword">Password</label>
+          <input
+            type="password"
+            id="newPassword"
+            name="newPassword"
+            placeholder="Password"
+            required />
+        </div>
+        <div class="input-group">
+          <label for="newPasswordConfirm">Confirm Password</label>
+          <input
+            type="password"
+            id="newPasswordConfirm"
+            name="newPasswordConfirm"
+            placeholder="Confirm Password"
+            required />
+        </div>
+
+        <button type="submit" id="createAccount">Create Account</button>
+      </form>
+
+      <div id="set">
+        <div id="count">Length : 0</div>
+        <i id="see" onclick="see()" class="far fa-eye"></i>
+      </div>
+      <div id="check0">
+        <i class="far fa-check-circle"></i> <span> Length more than 5.</span>
+      </div>
+      <div id="check1">
+        <i class="far fa-check-circle"></i> <span> Length less than 10.</span>
+      </div>
+      <div id="check2">
+        <i class="far fa-check-circle"></i>
+        <span> Contains numerical character.</span>
+      </div>
+      <div id="check3">
+        <i class="far fa-check-circle"></i>
+        <span>Contains special character.</span>
+      </div>
+      <div id="check4">
+        <i class="far fa-check-circle"></i>
+        <span>Shouldn't contain spaces.</span>
+      </div>
```

Comment:
These requirements are interesting. Does our game have a responsibility to enforce such rules on passwords? It seems reasonable to have minimum and maximum length requirements, but I don't think the rest is necessary.

Reply:
follow your decision.